### PR TITLE
Database/Website integration

### LINKED ISSUE #73: racing environment Mini-map
4.8 As a user I would like to be able to see the position of the other racers so that i can tell where I am relative to them during the race.
Acceptance criteria: The cars spawned are all visible to user on track and mini-map.

FR 1.22

### LINKED ISSUE #75: Make cars more visible on mini-map
Improvement to user story 4.8 "As a user I would like to be able to see the position of the other racers so that i can tell where I am relative to them during the race." (issue #73) to make the cars on the map more visible to the player

### LINKED ISSUE #86: Change the title of the 'machine' map
Race Against Machine isn't very good wordplay. If we don't use The in there we're losing fans.

### LINKED ISSUE #89: Implementation of difficulty choices on the frontend
There are currently 5 difficulties on the godot engine, we will add these options to the frontend

### PR DESCRIPTION
**Pull request by drakeaxelrod**
Added the ability to change difficulties for the race map, in addition
to the number of laps for a race. changed the angle of the arduino
camera. added some more stats and information to the arduino camera.
some general bug fixes.

### CODE REVIEW

File: godot-resources/mods/RaceAgainstMod/src/environments/raceagainst/Boost.gd [hunk 1]
```diff
@@ -1,20 +1,17 @@
...
-		var direction= Vector3(0,0,-5) #example: direction in -z
+	if body.has_method("_integrate_forces") && enable_boost:
+		var direction = boost_direction
 		var global_direction = body.global_transform.basis.xform(direction)
-		body.add_force(global_direction * forcemult, Vector3(0,0,0))
-		#for wheel in body._rightw:
-		#	wheel.spring_force = 100
-		#for wheel in body._leftw:
-		#	wheel.spring_force = 100
+		body.add_force(global_direction * power, Vector3(0,0,0))
```
Comment (Axel-J-L) [inline godot-resources/mods/RaceAgainstMod/src/environments/raceagainst/Boost.gd:17]: Very well thought of.

File: godot-resources/mods/RaceAgainstMod/src/environments/raceagainst/Coordinates.gd [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+# Called when the node enters the scene tree for the first time.
+func _ready():
+	pass # Replace with function body.
+
+
+# Called every frame. 'delta' is the elapsed time since the previous frame.
+func _process(delta):
+	print()
+	pass
```
Comment (Axel-J-L) [inline godot-resources/mods/RaceAgainstMod/src/environments/raceagainst/Coordinates.gd:17]: I realize this was for testing purposes. But should probably be removed on master.

Comment (Axel-J-L) [reply godot-resources/mods/RaceAgainstMod/src/environments/raceagainst/Coordinates.gd:17]: Never mind, realized it was to print coordinates.

File: server/index.js [hunk 3]
```diff
@@ -17,14 +17,23 @@ server.get("/", async (request, reply) => {
...
-  const sql =
-    "INSERT INTO results (username, time, course) VALUES ($1, $2, $3);";
-  const values = [
-    request.body.username,
-    request.body.time,
-    request.body.course,
-  ];
+  console.log(request.body.time);
+  const sql = "INSERT INTO racing (username, time, course, placement) VALUES ($1, $2, $3, $4);";
+  const values = [request.body.name, request.body.time, request.body.course, request.body.placement];
```
Comment (Axel-J-L) [inline server/index.js:36]: Great work.

**Review by Axel-J-L — APPROVED** [review]

File: server/index.js [hunk 4]
```diff
@@ -17,14 +17,23 @@ server.get("/", async (request, reply) => {
   reply.send(result.rows);
 });
 
+server.get("/godot", async (request, reply) => {
+  const sql = "SELECT * FROM settings;";
+  const result = await client.query(sql);
+  reply.send(result.rows);
+});
+
+server.put("/godot", async (request, reply) => {
```
Comment (VernitaJ) [inline server/index.js:26]: This will be altered slightly in #92  to ensure we use the correct entry.

**Pull request merged**